repl/src/repl.rs

```diff
@@ -362,17 +363,41 @@ fn set_globals(
             Ok(())
         }
         Pattern::Record { ref fields, .. } => {
-            let iter = fields
-                .iter()
-                .zip(::vm::dynamic::field_iter(&value, typ, vm));
-            for (field, (field_value, field_type)) in iter {
-                match field.value {
-                    Some(ref field_pattern) => {
-                        set_globals(vm, field_pattern, &field_type, &field_value)?
+            // FIXME: not sure if this is necessary, or if this has already
+            // been done. ::vm::dynamic::field_iter did this
+            // so I'm doing it too
+            let resolved_type = resolve::remove_aliases_cow(&*vm.global_env().get_env(), typ);
+
+            for pattern_field in fields.iter() {
+                let field_name: &Symbol = &pattern_field.name.value;
+                let field_value: RootedValue<&Thread> =
+                    value.get_field(field_name.declared_name())
+                    // FIXME: can I use the type checker for this error?
+                    // FIXME: apparently this error message isn't actually getting displayed. not sure why
```

Comment:
This should never happen as the typechecker does validate that the pattern is valid.

Reply:
It doesn't output any errors though:

```
>let {x} = {}
>
```

Reply:
😓 Should have checked what the code actually does after that FIXME. It appears it only compiles with the pattern if the binding is a function which is an error.

If you replace the `eval_expr` creation in `fn eval_line_` with this it will do the necessary rewrite for any type of pattern and actually check it.

```rust
            // We can't compile function bindings by only looking at `let_binding.expr`
            // so rewrite `let f x y = <expr>` into `let f x y = <expr> in f`
            let id = match unpack_pattern.value {
                Pattern::Ident(ref id) if !let_binding.args.is_empty() => {
                    id.clone()
                }
                _ => {
                    let id = Symbol::from("repl_temp");
                    let_binding.name = pos::spanned(let_binding.name.span, Pattern::As(id.clone(), Box::new(let_binding.name)));
                    TypedIdent::new(id)
                }
            };
            let id = pos::spanned2(0.into(), 0.into(), Expr::Ident(id.clone()));
            let expr = Expr::LetBindings(vec![let_binding], Box::new(id));
            let eval_expr = pos::spanned2(0.into(), 0.into(), expr);

```